soh/soh/Enhancements/randomizer/3drando/logic.cpp

```diff
@@ -553,9 +591,10 @@ namespace Logic {
     AdultsWallet    = ProgressiveWallet     >= 1;
     BiggoronSword   = BiggoronSword || ProgressiveGiantKnife >= 2;
 
-    ScarecrowSong    = ScarecrowSong || ctx->GetOption(RSK_SKIP_SCARECROWS_SONG) || (ChildScarecrow && AdultScarecrow);
-    Scarecrow        = Hookshot && CanPlay(ScarecrowSong);
-    DistantScarecrow = Longshot && CanPlay(ScarecrowSong);
+    //you need at least 2 buttons for scarecrow song
+    ScarecrowSong    = ScarecrowSong || (RSK_SKIP_SCARECROWS_SONG && Ocarina && OcarinaButtons >= 2) || (ChildScarecrow && AdultScarecrow);
```

Comment:
The `ctx->GetOption` piece seems to have disappeared from this line. So rather than checking against the value of the option, you're checking against the value of the RSK enum key, which will just always be true. Adding the missing `ctx->GetOption()` back in should fix that.

Reply:
Whoops. Fixed.